### ISSUE TITLE
Add tests using storage framework for the default storageclass

### ISSUE THREAD

**Issue by msau42** [kubernetes/kubernetes#69602]
**Is this a BUG REPORT or FEATURE REQUEST?**:
@kubernetes/sig-storage-feature-requests 

**What happened**:
We can consider these tests as a future conformance profile candidate.  See https://github.com/kubernetes/kubernetes/issues/65155#issuecomment-411486293

**Comment by ddebroy** [comment]
Is it typically expected that a certified kubernetes distro sets up a default storage class (perhaps based on the environment) once installed? 

I had submitted https://github.com/kubernetes/kubernetes/pull/68949 but in the context of the above, I wonder if the presence of a default storage class should always be assumed.

**Comment by msau42** [comment]
I don't think we can require that all distros provide a default storage class, however, we are looking into an optional conformance profile that includes storage tests.  It is expected that distros that conform to the conformance profile must provide a default storageclass out of the box that doesn't need to be configured by users.

**Comment by msau42** [comment]
So conformance tests have a limitation that they need to be defined as individual test cases, and not as part of a framework.  ie, the `ConformanceIt()` won't work with the current storage test framework because the same It() is run for plugin-specific tests.

I can see two ways to address this:
1. Define a new `StorageConformanceIt()` that can apply the conformance tag only if the driver type is "default storageclass"
2. Define separate suite for default storageclass tests, and refactor test cases to be able to share test case code. Only the `It()` definition is duplicated.
